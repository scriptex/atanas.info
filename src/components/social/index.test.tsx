import * as React from 'react';
import { render } from '@testing-library/react';

import { Social } from '.';
import * as shared from '../../scripts/shared';

jest.mock('../../data/last.fm-insights.json', () => ({
	default: {
		weeklyAlbumChart: [],
		topAlbums: []
	}
}));

jest.spyOn(shared, 'formatDate');

(shared.formatDate as jest.Mock).mockImplementation(() => 'Mock date');

describe('Social component', () => {
	it('Should render the Social component', () => {
		const { asFragment } = render(<Social />);

		expect(asFragment()).toMatchSnapshot();

		expect(asFragment()).toMatchSnapshot();
	});
});
