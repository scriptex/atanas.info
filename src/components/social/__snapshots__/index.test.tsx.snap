// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social component Should render the Social component 1`] = `
<DocumentFragment>
  <main
    class="o-main "
  >
    <section
      class="c-section c-section--social"
      id="social"
    >
      <header
        class="c-section__header"
      >
        <h2>
          Social
        </h2>
      </header>
      <div
        class="lines"
      >
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
      </div>
      <div
        class="lines"
      >
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
      </div>
      <div
        class="c-section__entry"
      >
        <div
          class="o-shell"
        >
          <div
            class="o-grid"
          >
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Codersrank Profile
              </h3>
              <codersrank-widget
                branding="false"
                username="scriptex"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                LinkedIn Profile
              </h3>
              <div
                class="badge-base LI-profile-badge linkedin-frame"
                data-locale="en_US"
                data-size="large"
                data-theme="dark"
                data-type="HORIZONTAL"
                data-vanity="scriptex"
                data-version="v1"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Github Profile
              </h3>
              <iframe
                class="github-frame"
                frameborder="0"
                height="225"
                loading="lazy"
                src="https://github.com/sponsors/scriptex/card"
                style="border: 0px;"
                title="Sponsor scriptex"
                width="600"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Twitter Profile
              </h3>
              <div />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                YouTube Channel
              </h3>
              <iframe
                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                allowfullscreen=""
                class="youtube-frame"
                frameborder="0"
                height="400"
                loading="lazy"
                src="https://www.youtube.com/embed/?list=PLgLu3XOevO8_G7n5n9Mmoe-xPwuwpGDYr"
                title="YouTube video player"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Spotify playlist
              </h3>
              <iframe
                allow="encrypted-media"
                class="spotify-frame"
                frameborder="0"
                height="380"
                loading="lazy"
                src="https://open.spotify.com/embed/playlist/37i9dQZF1EpzZomgLqaOnp?theme=0"
                width="100%"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Stackoverflow flair
              </h3>
              <div
                class="stackoverflow-frame"
              >
                <a
                  href="https://stackoverflow.com/users/4140082/atanas-atanasov"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <img
                    alt="profile for Atanas Atanasov at Stack Overflow, Q&A for professional and enthusiast programmers"
                    height="58"
                    src="https://stackoverflow.com/users/flair/4140082.png"
                    title="profile for Atanas Atanasov at Stack Overflow, Q&A for professional and enthusiast programmers"
                    width="208"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <header
        class="c-section__header"
      >
        <h2>
          <a
            href="https://www.last.fm/user/scriptex"
            rel="noopener noreferrer"
            target="_blank"
          >
            Last.FM
          </a>
           statistics
          <small>
            Updated at Mock date
          </small>
        </h2>
      </header>
      <div
        class="c-section__entry"
      >
        <div
          class="o-shell"
        >
          <div
            class="o-grid"
          >
            <div
              class="o-grid__item xs-12"
            >
              <h3>
                Top albums for last week:
              </h3>
              <div
                class="carousel"
              >
                <div
                  class="carousel__container"
                  style="transform: translateZ(NaNpx) rotateY(NaNdeg);"
                />
              </div>
              <div
                class="carousel__controls"
              >
                <button
                  class="carousel__control carousel__control--prev"
                >
                  Previous
                </button>
                <button
                  class="carousel__control carousel__control--next"
                >
                  Next
                </button>
              </div>
            </div>
            <div
              class="o-grid__item xs-12"
            >
              <h3>
                Top albums for last month:
              </h3>
              <div
                class="carousel"
              >
                <div
                  class="carousel__container"
                  style="transform: translateZ(NaNpx) rotateY(NaNdeg);"
                />
              </div>
              <div
                class="carousel__controls"
              >
                <button
                  class="carousel__control carousel__control--prev"
                >
                  Previous
                </button>
                <button
                  class="carousel__control carousel__control--next"
                >
                  Next
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn "
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn "
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
            </a>
          </li>
        </ul>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`Social component Should render the Social component 2`] = `
<DocumentFragment>
  <main
    class="o-main "
  >
    <section
      class="c-section c-section--social"
      id="social"
    >
      <header
        class="c-section__header"
      >
        <h2>
          Social
        </h2>
      </header>
      <div
        class="lines"
      >
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
      </div>
      <div
        class="lines"
      >
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
      </div>
      <div
        class="c-section__entry"
      >
        <div
          class="o-shell"
        >
          <div
            class="o-grid"
          >
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Codersrank Profile
              </h3>
              <codersrank-widget
                branding="false"
                username="scriptex"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                LinkedIn Profile
              </h3>
              <div
                class="badge-base LI-profile-badge linkedin-frame"
                data-locale="en_US"
                data-size="large"
                data-theme="dark"
                data-type="HORIZONTAL"
                data-vanity="scriptex"
                data-version="v1"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Github Profile
              </h3>
              <iframe
                class="github-frame"
                frameborder="0"
                height="225"
                loading="lazy"
                src="https://github.com/sponsors/scriptex/card"
                style="border: 0px;"
                title="Sponsor scriptex"
                width="600"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Twitter Profile
              </h3>
              <div />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                YouTube Channel
              </h3>
              <iframe
                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                allowfullscreen=""
                class="youtube-frame"
                frameborder="0"
                height="400"
                loading="lazy"
                src="https://www.youtube.com/embed/?list=PLgLu3XOevO8_G7n5n9Mmoe-xPwuwpGDYr"
                title="YouTube video player"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Spotify playlist
              </h3>
              <iframe
                allow="encrypted-media"
                class="spotify-frame"
                frameborder="0"
                height="380"
                loading="lazy"
                src="https://open.spotify.com/embed/playlist/37i9dQZF1EpzZomgLqaOnp?theme=0"
                width="100%"
              />
            </div>
            <div
              class="o-grid__item xs-12 sm-6"
            >
              <h3>
                Stackoverflow flair
              </h3>
              <div
                class="stackoverflow-frame"
              >
                <a
                  href="https://stackoverflow.com/users/4140082/atanas-atanasov"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <img
                    alt="profile for Atanas Atanasov at Stack Overflow, Q&A for professional and enthusiast programmers"
                    height="58"
                    src="https://stackoverflow.com/users/flair/4140082.png"
                    title="profile for Atanas Atanasov at Stack Overflow, Q&A for professional and enthusiast programmers"
                    width="208"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <header
        class="c-section__header"
      >
        <h2>
          <a
            href="https://www.last.fm/user/scriptex"
            rel="noopener noreferrer"
            target="_blank"
          >
            Last.FM
          </a>
           statistics
          <small>
            Updated at Mock date
          </small>
        </h2>
      </header>
      <div
        class="c-section__entry"
      >
        <div
          class="o-shell"
        >
          <div
            class="o-grid"
          >
            <div
              class="o-grid__item xs-12"
            >
              <h3>
                Top albums for last week:
              </h3>
              <div
                class="carousel"
              >
                <div
                  class="carousel__container"
                  style="transform: translateZ(NaNpx) rotateY(NaNdeg);"
                />
              </div>
              <div
                class="carousel__controls"
              >
                <button
                  class="carousel__control carousel__control--prev"
                >
                  Previous
                </button>
                <button
                  class="carousel__control carousel__control--next"
                >
                  Next
                </button>
              </div>
            </div>
            <div
              class="o-grid__item xs-12"
            >
              <h3>
                Top albums for last month:
              </h3>
              <div
                class="carousel"
              >
                <div
                  class="carousel__container"
                  style="transform: translateZ(NaNpx) rotateY(NaNdeg);"
                />
              </div>
              <div
                class="carousel__controls"
              >
                <button
                  class="carousel__control carousel__control--prev"
                >
                  Previous
                </button>
                <button
                  class="carousel__control carousel__control--next"
                >
                  Next
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn "
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn "
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
            </a>
          </li>
        </ul>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
